text/3389-manifest-lint.md

```diff
@@ -0,0 +1,398 @@
+  `rustc`
+
+To know which table under `[lints]` a particular lint belongs under, it is the part before `::` in the lint
+name.  If there isn't a `::`, then the tool is `rust`.  For example a warning
+about `unsafe` would be `lints.rust.unsafe` but a lint about
+`clippy::enum_glob_use` would be `lints.clippy.enum_glob_use`.
+
+## The `lints` table
+
+*as a new [`[workspace]` entry](https://doc.rust-lang.org/cargo/reference/workspaces.html#the-workspace-section)*
+
+The `workspace.lints` table is where you define lint configuration to be inherited by members of a workspace.
+
+Specifying a workspace lint configuration is similar to package lints.
+
+Example:
+
+```toml
+# [PROJECT_DIR]/Cargo.toml
+[workspace]
+members = ["crates/*"]
+
+[workspace.lints.rust]
+unsafe = "forbid"
+```
+
+```toml
+# [PROJECT_DIR]/crates/bar/Cargo.toml
+[package]
+name = "bar"
+version = "0.1.0"
+
+[lints]
+workspace = true
+```
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+When parsing a manifest, cargo will resolve workspace inheritance for
+`lints.workspace = true` as it does with basic fields, when `workspace` is
+present, no other fields are allowed to be present.  This precludes having the
+package override the workspace on a lint-by-lint basis.
+
+cargo will contain a mapping of tool to underlying command (e.g. `rust` to
+`rustc`, `clippy` to `rustc` when clippy is the driver, `rustdoc` to
+`rustdoc`).  When running the underlying command for the specified package,
+cargo will transform the lints from `lint = level` to `--level lint`, sort them
+by priority and then lint name, and pass them on the command line before other
+configuration, `RUSTFLAGS`, allowing user configuration to override package
+configuration.  These flags will be fingerprinted so changing them will cause a
+rebuild only for the commands where they are used.
+
+Note that this means that `[lints]` does not affect dependencies which is
+normally not an issue due to `--cap-lints` being used for dependencies.
+
+Initially, the only supported tools will be:
+- `rust`
+- `clippy`
+- `rustdoc`
+
+Addition of third-party tools would fall under their
+[attributes for tools](https://github.com/rust-lang/rust/issues/44690).
+
+**Note:** This reserves the tool name `workspace` to allow workspace inheritance.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Since `[lints]` only affects the associated package, and not dependencies, it
+will not work with `future-incompat` lints that are meant to be applied to
+dependencies.  This may cause some user confusion.
+
+There has been some user/IDE confusion about running commands like `rustfmt`
+directly and expecting them to pick up configuration only associated with their
+higher-level cargo-plugins despite that configuration (like `package.edition`)
+being cargo-specific.  By baking the configuration for rustc, rustdoc, and
+clippy directly into cargo, we will be seeing more of this.  A hope is that
```

Comment:
155776772dbcf834c93c5d752f5eae381430d831 clarifies that I meant configured lint levels, not linter configuration.  This section came at the behest of rustfmt's maintainer because they have run into user confusion over the behavior difference between running `rustfmt` directly and `cargo fmt`.

Translated to this RFC, this would be like someone running rustc directly and being confused as to why their `Cargo.toml` lint levels were not applied.  As I said, I don't expect this to be a big deal but I went ahead and called out this potential confusion.